pkg/sriov/networknodestate.go

```diff
@@ -142,3 +145,62 @@ func (builder *NetworkNodeStateBuilder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WaitUntilSyncStatus waits for the duration of the defined timeout or until the
+// SriovNetworkNodeState gets to a specific syncStatus.
+func (builder *NetworkNodeStateBuilder) WaitUntilSyncStatus(syncStatus string, timeout time.Duration) error {
+	glog.V(100).Infof("Waiting for the defined period until SriovNetworkNodeState %s has syncStatus %s",
+		builder.Objects.Name, syncStatus)
+
+	if builder.errorMsg != "" {
+		return fmt.Errorf(builder.errorMsg)
+	}
+
+	if syncStatus == "" {
+		return fmt.Errorf("syncStatus can't be empty")
+	}
+
+	// Polls every retryInterval to determine if SriovNetworkNodeState is in desired syncStatus.
+	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		err := builder.Discover()
+
+		if err != nil {
+			return false, nil
+		}
+
+		if builder.Objects.Status.SyncStatus == syncStatus {
```

Comment:
can be simplified to 
```suggestion
                return builder.Objects.Status.SyncStatus == syncStatus, nil
```